olive/engine/engine.py

```diff
@@ -380,6 +388,11 @@ def run(
             except Exception as e:
                 logger.warning(f"Failed to run Olive on {accelerator_spec}: {e}", exc_info=True)
 
+            if origin_target_system.olive_managed_env:
+                self.target.remove()
```

Comment:
>self.target.remove() [](http://example.com/codeflow?start=16&length=20)

do we need restore the target to original one?

Reply:
modified